### DISCUSSION TITLE
UI Image sprite 변경 오류

### DISCUSSION THREAD

**Discussion by hwisulee** [naverz/zepeto-studio-kor#1233] [5. Q&A]
![image](https://user-images.githubusercontent.com/62528282/186787779-91e4568e-6c13-406a-91f4-b272e9287fbd.png)

다음과 같이 게임에서 물고기 버튼을 누르면 낚시 게임이 작동할 수 있도록 제작 중에 있습니다.
낚시 게임은 stardew velly라는 게임의 낚시 시스템과 유사한 방식으로 동작하도록 설계 중에 있습니다.

![image](https://user-images.githubusercontent.com/62528282/186788706-74b36f18-dc31-4fb6-9fef-a8652fdf3e08.png)

![image](https://user-images.githubusercontent.com/62528282/186787895-4457108f-263a-423b-ba6d-579d60713edd.png)

오류가 나는 코드는 위와 같습니다.
다른 함수에서 csv 파일을 읽은 뒤, interface에서 정해준 형식대로 split 시킨 데이터를 배열에 넣어주었습니다.
이 후, 테스트를 위해 데이터 하나의 spriteID 부분을 읽어와 log로 띄웠을 때 정상적으로 나오는 것을 확인하고 GetComponent<Image>()를 사용해 spirte를 변경해주는 과정을 수행하려고 했습니다.

그러나 다음과 같은 오류가 발생하였습니다.

![image](https://user-images.githubusercontent.com/62528282/186787935-7db84add-4da4-44c9-b033-35654a3d9191.png)

혹시 제페토에서 지원하지 않는 사항인가요?

**Comment by YSHFM** [comment]
![image](https://user-images.githubusercontent.com/106574541/186798564-84e2673b-7c14-4d3e-9857-dc192b86efbe.png)
저는 정상적으로 동작됩니다. 
![image](https://user-images.githubusercontent.com/106574541/186798787-1891bb0f-38e7-4cfd-9296-b48a2e447916.png)
같은 코드로 컴파일된 js코드입니다.

정상적으로 삽입되는 걸로 봐서는 Resources.Load의 로드된 데이터가 의도한 데이터가 정확하게 맞는 지 먼저 확인 해보셔야 할 것 같습니다.

**Reply by hwisulee** [reply]
![image](https://user-images.githubusercontent.com/62528282/186806174-7a081220-e359-40f2-9c88-6f99c4e5d525.png)


#544 에서 발생했던 문제와 동일한 것 같습니다.
일단 무엇이 문제였는지는 덕분에 파악했습니다. 감사합니다.
